File: client/client/Services/IOfferService.cs
using client.Dto;
using client.Dto.Offer;
using client.Pages;

namespace client.Services;

public interface IOfferService
{
    public Task<OfferResponseDto> CreateOfferAsync(CreateOfferRequestDto createOfferRequestDto);
    public Task<List<OfferResponseDto>> GetOffersAsync(FilterRequestDto? filterRequestDto);
    public Task<OfferResponseDto> GetOfferByIdAsync(string id);
    public Task<List<CategoryDto>> GetCategoriesAsync();
    public Task<List<OfferResponseDto>> GetOffersByBusinessIdAsync(string id);
}

File: server/src/main/java/com/example/server/services/IOfferService.java
package com.example.server.services;

import com.example.server.dto.offer.CreateOfferRequestDto;
import com.example.server.dto.offer.FilterRequestDto;
import com.example.server.dto.offer.OfferResponseDto;
import com.example.server.dto.offer.UpdateOfferRequestDto;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;
import java.util.Optional;

public interface IOfferService
{
  @Transactional OfferResponseDto createOffer(
      CreateOfferRequestDto offerRequestDto, String userId);
  OfferResponseDto getOfferById(String id);
  List<OfferResponseDto> getOffers(FilterRequestDto filterRequestDto);
  List<OfferResponseDto> getOffersByBusinessId(String businessId);
}
